cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -81,3 +80,132 @@ func (s *Server) GetPackageRepositories(ctx context.Context, request *corev1.Get
 		Repositories: responseRepos,
 	}, nil
 }
+
+// GetAvailablePackages streams the available packages based on the request.
+func (s *Server) GetAvailablePackages(ctx context.Context, request *corev1.GetAvailablePackagesRequest) (*corev1.GetAvailablePackagesResponse, error) {
+	log.Infof("+GetAvailablePackages(namespace=[%s])", request.Namespace)
+	repos, err := getHelmRepos(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	responsePackages := []*corev1.AvailablePackage{}
+	for _, repoUnstructured := range repos.Items {
+		name, found, err := unstructured.NestedString(repoUnstructured.Object, "metadata", "name")
+		if err != nil || !found {
+			log.Errorf("required field metadata.name not found on HelmRepository: %w:\n%v", err, repoUnstructured.Object)
```

Comment:
If this ever happened (not sure it's possible, given the field here is `metadata.name`), wouldn't we want an internal error response?

Reply:
i don't know. i thought i'd log the error and move on to next repo. Kind of mute either way, because it'll probably never happen